frontend/src/modules/userManagement/collections/userCollection.js

```diff
@@ -2,13 +2,134 @@
 define(function(require) {
   var Backbone = require('backbone');
   var UserModel = require('../models/userModel');
+  var Moment = require('moment');
 
   var UserCollection = Backbone.Collection.extend({
     model: UserModel,
-    comparator: function(model) {
-      return model.get('email').toLowerCase();
+
+    sortBy: 'email',
+    direction: 1,
+    mailSearchTerm: false,
+
+    filterGroups: {},
+
+    lastAccess: null,
+
+    initialize: function() {
+      this.listenTo(this, 'sync', this.onSync);
+    },
+
+    comparator: function(ma, mb) {
+      var a = ma.get(this.sortBy);
+      var b = mb.get(this.sortBy);
+
+      if (Array.isArray(a) && Array.isArray(b)) {
+        a = a[0];
+        b = b[0];
+      }
+
+      if (this.sortBy === 'lastAccess') {
+        a = new Date(a || '01.01.1900');
+        b = new Date(b || '01.01.1900');
+      }
+
+      if (typeof a === 'string' && typeof b === 'string') {
+        a = a.toLowerCase();
+        b = b.toLowerCase();
+      }
+
+      if (a > b) return this.direction;
+      else if (a < b) return this.direction*-1;
+      return 0;
+    },
+
+    url: 'api/user',
+
+    updateFilter: function(filterMap) {
+      this.filterGroups = filterMap;
+      this.sortCollection();
+    },
+
+    sortCollection: function() {
+      this.resetHidden();
+      this.filter();
+      this.searchByMail();
+      this.sort();
+    },
+
+    filter: function() {
+      this.models.forEach(function(model) {
+        this.filterFailedLoginCount(model);
+        this.filterTenants(model);
+        this.filterRoleNames(model);
+        this.filterLastAccess(model);
+      }, this);
     },
-    url: 'api/user'
+
+    filterFailedLoginCount: function(model) {
+      if (this.filterGroups.failedLoginCount
+        && this.filterGroups.failedLoginCount.indexOf(model.get('failedLoginCount').toString()) < 0) {
+        model.set('_isHidden', true);
+      }
+    },
+
+    filterTenants: function() {
+      if (this.filterGroups.tenantName
+        && this.filterGroups.tenantName.indexOf(model.get('tenantName')) < 0) {
```

Comment:
Getting the error `Uncaught ReferenceError: model is not defined  userCollection.js:78` when filtering or sorting by tenant.